src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -20800,6 +20799,62 @@ static void F(B? b)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "b").WithArguments("a", "A.explicit operator S(A a)").WithLocation(13, 17));
         }
 
+        // PROTOTYPE(NullableReferenceTypes): Should only report one warning for: A a = (B)s;
```

Comment:
I didn't understand the comment. We only have a user-defined conversion that may return `null`, and we're assigning to a non-null local in all three cases. I'd think all three assignments should warn, and all three dereferences should warn too.

Reply:
Sorry, the comment was not precise. It meant we should only report one `WRN_ConvertingNullableToNonNullable` warning in each case, as you'd suggested below. I've updated the comment.

---
In reply to: [189129614](https://github.com/dotnet/roslyn/pull/26356#discussion_r189129614) [](ancestors = 189129614)